File: leap_year.js
#!/usr/bin/env node
// check whether a year's a leap year

function isLeapYear(y) {
  return y % 400 === 0 ? true : (y % 4 === 0 && y % 100 !== 0) ? true : false
}

console.log(isLeapYear(2000));
console.log(isLeapYear(1900));
console.log(isLeapYear(1800));
console.log(isLeapYear(1996));
console.log(isLeapYear(1999));

File: leap_year.py
#!/usr/bin/env python

def leap_year(year):
  if year % 400 == 0:
    return True
  elif year % 4 == 0 and year % 100 != 0:
    return True
  else:
    return False

for y in [2000,1900,1800,1996,1999]:
  print str(y) + ": " + str(leap_year(y))
  